src/app/games/lodestone/LodestoneDummyMinimax.ts

```diff
@@ -0,0 +1,97 @@
+        });
+    }
+    private captureCombinations(state: LodestoneState, numberOfCaptures: number): MGPSet<LodestoneCaptures> {
+        if (numberOfCaptures === 0) {
+            return new MGPSet([{ top: 0, bottom: 0, left: 0, right: 0 }]);
+        } else {
+            const combinations: MGPSet<LodestoneCaptures> = new MGPSet();
+            const available: LodestoneCaptures = state.remainingSpacesDetails();
+            const subCombinations: MGPSet<LodestoneCaptures> = this.captureCombinations(state, numberOfCaptures-1);
+            for (const subCombination of subCombinations) {
+                if (subCombination.top + 1 <= available.top) {
+                    combinations.add({ ...subCombination, top: subCombination.top + 1 });
+                }
+                if (subCombination.bottom + 1 <= available.bottom) {
+                    combinations.add({ ...subCombination, bottom: subCombination.bottom + 1 });
+                }
+                if (subCombination.left + 1 <= available.left) {
+                    combinations.add({ ...subCombination, left: subCombination.left + 1 });
+                }
+                if (subCombination.right + 1 <= available.right) {
+                    combinations.add({ ...subCombination, right: subCombination.right + 1 });
+                }
+            }
+            if (combinations.size() === 0) {
+                // It was not possible to place all captures, so we keep the sub combinations
+                return subCombinations;
+            }
+            return combinations;
+        }
+    }
+    private flatMapEmptyCoords(state: LodestoneState, f: (coord: Coord) => LodestoneMove[]): LodestoneMove[] {
+        let moves: LodestoneMove[] = [];
+        for (let y: number = 0; y < LodestoneState.SIZE; y++) {
+            for (let x: number = 0; x < LodestoneState.SIZE; x++) {
+                const coord: Coord = new Coord(x, y);
+                const piece: LodestonePiece = state.getPieceAt(coord);
+                if (piece.isEmpty() && piece.isUnreachable() === false) {
+                    moves = moves.concat(f(coord));
+                } else if (piece.isLodestone() && piece.owner === state.getCurrentPlayer()) {
+                    // A player can also put a lodestone on its previous location
+                    moves = moves.concat(f(coord));
+                }
+            }
+        }
+        return moves;
+    }
+    private nextDirection(state: LodestoneState): LodestoneDirection[] {
+        const nextDirection: MGPOptional<LodestoneDirection> = state.nextLodestoneDirection();
+        if (nextDirection.isPresent()) {
+            return [nextDirection.get()];
+        } else {
+            return ['push', 'pull'];
+        }
+    }
+    public getBoardValue(node: LodestoneNode): NodeUnheritance {
+        const scores: [number, number] = node.gameState.getScores();
+        let score: number;
+        if (scores[0] === 24 && scores[1] === 24) {
```

Comment:
same behavior as the final else ?

Reply:
vrai, branche supprimée donc !

Reply:
abah non ! en enlevant cette branche, ça change tout (vu que si scores = [24, 24], on passe dans la branche qui fait gagner joueur 1 alors). Regarde le refactor proposé si c'est mieux, mais j'suis pas convaincu que c'est tellement mieux !